Fix compiler warning from HeliumAnalyserEfficiencyTest

The `HeliumAnalyserEfficiencyTest` is producing the following compiler warnings:

```
10:32:27 /Users/mantidbuilder/Jenkins/workspace/pull_requests-conda-osx/Framework/Algorithms/test/PolarizationCorrections/HeliumAnalyserEfficiencyTest.h:134:30: warning: implicit conversion from 'const size_t' (aka 'const unsigned long') to 'double' may lose precision [-Wimplicit-int-float-conversion]
10:32:27       x[i] = 2.0 + i * 8.0 / numBins;
10:32:27                            ~ ^~~~~~~
10:32:27 /Users/mantidbuilder/Jenkins/workspace/pull_requests-conda-osx/Framework/Algorithms/test/PolarizationCorrections/HeliumAnalyserEfficiencyTest.h:134:20: warning: implicit conversion from 'size_t' (aka 'unsigned long') to 'double' may lose precision [-Wimplicit-int-float-conversion]
10:32:27       x[i] = 2.0 + i * 8.0 / numBins;
10:32:27                    ^ ~
10:32:27 2 warnings generated.
```

These should be resolved, and need to be done ASAP to allow the pipeline threshold to be changed so that PRs that introduce new warnings will fail.

--- a/Framework/Algorithms/test/PolarizationCorrections/HeliumAnalyserEfficiencyTest.h
+++ b/Framework/Algorithms/test/PolarizationCorrections/HeliumAnalyserEfficiencyTest.h
@@ -131,7 +131,7 @@ private:
     std::vector<double> yNsf(numBins);
     std::vector<double> ySf(numBins);
     for (size_t i = 0; i < numBins; ++i) {
-      x[i] = 2.0 + i * 8.0 / numBins;
+      x[i] = 2.0 + static_cast<double>(i) * 8.0 / static_cast<double>(numBins);
       yNsf[i] = 0.9 * std::exp(-0.0733 * x[i] * 12 * (1 - examplePHe));
       ySf[i] = 0.9 * std::exp(-0.0733 * x[i] * 12 * (1 + examplePHe));
     }

